rest-service/manager_rest/rest/resources_v3_1/agents.py

```diff
@@ -43,3 +48,77 @@ def get(self, pagination=None):
             node_ids=args.get('node_ids'),
             node_instance_ids=args.get('node_instance_ids'),
             install_method=args.get('install_methods'))
+
+
+class AgentsName(SecuredResource):
+    @rest_decorators.exceptions_handled
+    @rest_decorators.marshal_with(models.Agent)
+    @authorize('agent_get')
+    def get(self, name):
+        """
+        Get agent by name
+        """
+        validate_inputs({'name': name})
+        return get_storage_manager().get(models.Agent, name)
+
+    @rest_decorators.exceptions_handled
+    @rest_decorators.marshal_with(models.Agent)
+    @authorize('agent_create')
+    def put(self, name):
```

Comment:
Why put instead of post? Put is usually used when you create or update in the same command. I think post is better suited here.

Reply:
I'll change to post